### PR TITLE
many: use ResolvedSource() from gadget content when writing boot assets

### PR DESCRIPTION
**Pull request by mvo5**
This changes the gadget install and update code to use the "ResolvedSource()" in a gadget update or install. This is a preview at this point and it needs #9777 and has some ugliness still that needs discussion/addressing. Note that it does not yet do a gadget assets update when a new kernel is installed, that requires more work.

Not sure if this could be split up further.

### CODE REVIEW

File: gadget/layout.go [hunk 1]
```diff
@@ -243,29 +243,50 @@ func LayoutVolume(gadgetRootDir string, volume *Volume, constraints LayoutConstr
 	return vol, nil
 }
 
+type multiErr []error
+
+func (me multiErr) String() string {
+	s := ""
```
Comment (pedronis) [inline gadget/layout.go:249]: this is odd as we discussed in a PR that did something similar

File: gadget/layout.go [hunk 2]
```diff
@@ -243,29 +243,50 @@ func LayoutVolume(gadgetRootDir string, volume *Volume, constraints LayoutConstr
...
+	return s
+}
+
 // ResolveContentPaths resolves any "$kernel:" refs in the gadget
 // content and populates VolumeContent.resolvedSource with absolute
 // paths.
 //
 // XXX: maybe move into LayoutVolume(), operator on *Volume and make private?
-func ResolveContentPaths(lv *LaidOutVolume, gadgetRootDir, kernelRootDir string) error {
+func ResolveContentPaths(lv *Volume, gadgetRootDir, kernelRootDir string) error {
```
Comment (pedronis) [inline gadget/layout.go:264]: why this change?

**Review by pedronis — COMMENTED** [review]
there's quite a bit of rename and argument churn in this PR which doesn't make it very easy to review, maybe some can be avoided or the PR split ?

File: gadget/layout.go (hunk 2 above)
Comment (pedronis) [reply gadget/layout.go:264]: I also wonder if either Volume or LaidOutVolume should carry a flag to know whether it was resolved or not, we could use that early in functions that must operate only on resolved volumes

**Comment by mvo5** [comment]
Closing as it's a bit hard to review
